dojo/tools/rusty_hog/parser.py

```diff
@@ -19,14 +19,7 @@ def get_findings(self, json_output, test):
         return self.get_items(tree, test)
 
     def parse_json(self, json_output):
-        try:
-            data = json_output.read()
-            try:
-                tree = json.loads(str(data, 'utf-8'))
-            except:
-                tree = json.loads(data)
-        except:
-            raise Exception("Invalid format")
+        tree = json.loads(json_output.read())
```

Comment:
```suggestion
        tree = json.load(json_output)
```

Reply:
I have tested it without ".read()", but it throws exceptions:

uwsgi_1         |   File "/app/./dojo/tools/rusty_hog/parser.py", line 22, in parse_json
uwsgi_1         |     tree = json.loads(json_output)
uwsgi_1         |   File "/usr/local/lib/python3.8/json/__init__.py", line 341, in loads
uwsgi_1         |     raise TypeError(f'the JSON object must be str, bytes or bytearray, '
uwsgi_1         | TypeError: the JSON object must be str, bytes or bytearray, not TemporaryUploadedFile

Thus, I would keep it this way.

Reply:
No you must use `json.load(json_output)` format (no `s` wich mean string).
This calls a native implementation that is aware of the UTF8/16/32 situation.
https://github.com/python/cpython/blob/a6a088548063226233c08b8d35dde130746fdd10/Lib/json/__init__.py#L333-L341

Reply:
Ah, nice catch.